### PR TITLE
Parse complex expressions in parameters after * and **

### LINKED ISSUE #2: Cannot parse `f(*[] or [])`

### PR DESCRIPTION
**Pull request by zsol**
Allow `test`s after * and **, instead of only `expr`s in lib2to3's grammar. Fixes #2

### CODE REVIEW

File: black.py [hunk 1]
```diff
@@ -1255,7 +1255,7 @@ def whitespace(leaf: Leaf) -> str:  # noqa C901
                     # that, too.
                     return prevp.prefix
 
-        elif prevp.type == token.DOUBLESTAR:
+        elif prevp.type in {token.DOUBLESTAR, token.STAR}:
```
Comment (ambv) [inline black.py]: `prevp.type in STARS`

File: blib2to3/Grammar.txt [hunk 2]
```diff
@@ -138,8 +138,8 @@ arglist: argument (',' argument)* [',']
 # that precede iterable unpackings are blocked; etc.
 argument: ( test [comp_for] |
             test '=' test |
-	    '**' expr |
-	    star_expr )
+	    '**' test |
+            '*' test )
```
Comment (ambv) [inline blib2to3/Grammar.txt]: Yup, that's what CPython does.

Comment (zsol) [reply blib2to3/Grammar.txt]: where does this happen in CPython?

Comment (ambv) [reply blib2to3/Grammar.txt]: https://github.com/python/cpython/blob/master/Grammar/Grammar#L136-L139

File: black.py [hunk 3]
```diff
@@ -1364,7 +1364,7 @@ def whitespace(leaf: Leaf) -> str:  # noqa C901
             if not prevp or prevp.type == token.LPAR:
                 return NO
 
-        elif prev.type == token.EQUAL or prev.type == token.DOUBLESTAR:
+        elif prev.type in {token.EQUAL} | STARS:
```
Comment (ambv) [inline black.py]: You could have used your newly found power:
```
elif prev.type in {*STARS, token.EQUAL}:
```
😉

**Comment by ambv** [comment]
In the future merge by squashing. Merges look bad in history.

**Pull request merged**